pinot-core/src/test/java/org/apache/pinot/core/startree/v2/BaseStarTreeV2Test.java

```diff
@@ -91,32 +91,35 @@ abstract class BaseStarTreeV2Test<R, A> {
 
   private static final int NUM_SEGMENT_RECORDS = 100_000;
   private static final int MAX_LEAF_RECORDS = RANDOM.nextInt(100) + 1;
-  private static final String DIMENSION_D1 = "d1";
-  private static final String DIMENSION_D2 = "d2";
+  private static final String DIMENSION_D1 = "d1__COLUMN_NAME";
```

Comment:
Let's add some comments on why choosing this strange dimension name

Reply:
ack

Reply:
TBH I don't like this tests. It is very chaotic and difficult to read. Having to apply these modifications in order to test that names with underscores are not failing makes no sense. We should have a simple test method testing that condition.

But it can be done in a future PR.